### PR TITLE
sql: update auto_random document

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

Add some details to the document of auto_random.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v4.0 changes for now)
- [x] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/2478
- Other reference link(s):

### CODE REVIEW

File: reference/sql/attributes/auto-random.md [hunk 1]
```diff
@@ -50,12 +50,12 @@ create table t (a int auto_random, b varchar(255), primary key (a))
...
++ If the `INSERT` statement explicitly specifies the values of the integer primary key column, TiDB saves these values, which works similarly to the `AUTO_INCREMENT` attribute. Note that if you do not set `NO_AUTO_VALUE_ON_ZERO` in the `@@sql_mode` system variable, TiDB will automatically assign values to this column even if you explicitly specify the value of the integer primary key column as `0`.
 
 TiDB automatically assigns values in the following way:
 
-The highest five digits of the row value in binary (namely, shard bits) are determined by the starting time of the current transaction. The remaining digits are assigned values in an auto-increment order.
+The second highest five digits (no matter the highest digit is signed or unsigned) of the row value in binary (namely, shard bits) are determined by the starting time of the current transaction. The remaining digits are assigned values in an auto-increment order.
```
Comment (tangenta) [inline reference/sql/attributes/auto-random.md]:
```suggestion
The second highest five digits (no matter the column type is signed or unsigned) of the row value in binary (namely, shard bits) are determined by the starting time of the current transaction. The remaining digits are assigned values in an auto-increment order.
```

File: reference/sql/attributes/auto-random.md [hunk 2]
```diff
@@ -50,12 +50,12 @@ create table t (a int auto_random, b varchar(255), primary key (a))
...
++ If the `INSERT` statement explicitly specifies the values of the integer primary key column, TiDB saves these values, which works similarly to the `AUTO_INCREMENT` attribute. Note that if you do not set `NO_AUTO_VALUE_ON_ZERO` in the `@@sql_mode` system variable, TiDB will automatically assign values to this column even if you explicitly specify the value of the integer primary key column as `0`.
 
 TiDB automatically assigns values in the following way:
 
-The highest five digits of the row value in binary (namely, shard bits) are determined by the starting time of the current transaction. The remaining digits are assigned values in an auto-increment order.
+The second highest five digits (no matter the column type is signed or unsigned) of the row value in binary (namely, shard bits) are determined by the starting time of the current transaction. The remaining digits are assigned values in an auto-increment order.
```
Comment (yikeke) [inline reference/sql/attributes/auto-random.md]: 看原文的意思，(no matter the column type is signed or unsigned) 这个应该修饰“最高位”？如果是的话，可能需要把最高位也翻译出来。麻烦和作者再确认下～ @TomShawn

File: reference/sql/attributes/auto-random.md [hunk 3]
```diff
@@ -67,7 +67,48 @@ create table t (a int primary key auto_random(3), b varchar(255))
...
+
+{{< copyable "sql" >}}
+
+```sql
+insert into t (b) values ("b")
+select * from t;
+select last_insert_id()
+```
+
+You might see the following results:
```
Comment (yikeke) [inline reference/sql/attributes/auto-random.md]:
```suggestion
You might see the following result:
```

File: reference/sql/attributes/auto-random.md (hunk 2 above)
Comment (TomShawn) [reply reference/sql/attributes/auto-random.md]: @tangenta PTAL, thanks!

Comment (tangenta) [reply reference/sql/attributes/auto-random.md]: 这里没问题。如果列类型是有符号 (signed) 的，最高位为符号位，0 表示正数，1 表示负数；如果列类型是无符号 (unsigned) 的，最高位和其他位一样，没有特殊含义。

但是 auto_random 的实现是把最高位忽略掉，不论列类型是 signed 还是 unsigned，都从次高位算起，因此这里强调了一下。

Comment (TomShawn) [reply reference/sql/attributes/auto-random.md]: 好的，thx！

**Pull request merged**